@vmswr/plot.m

```diff
@@ -25,225 +24,226 @@
-			if(OldMarkerFormat)
-				dIdx = diff(obj.data.trialIndices,1,2);
-			else
-				dIdx = obj.data.trialIndices(:,3) - obj.data.trialIndices(:,1);
-			end
-			% find longest trial
-			mIdx = max(dIdx);
-			% find number of time bins in the spectrogram that corresponds to
-			spTimeStep = Args.TFfftWindow - Args.TFfftOverlap;
-			spTimeBins = floor(mIdx/spTimeStep) - Args.TFfftOverlap/spTimeStep;
-			% create matrix
-			nFreqs = (Args.TFfftPoints/2)+1;			
-			ops = zeros(nFreqs,spTimeBins);
-			opsCount = ops;
-			for ti = 1:obj.data.numSets
-				tftIdx = obj.data.trialIndices(ti,:);
-				if(OldMarkerFormat)
-					tfidx = tftIdx(1):tftIdx(2);
-				else
-					tfidx = tftIdx(1):tftIdx(3);
-				end
-				data = obj.data.analogData(tfidx);
-				if(~isempty(Args.RemoveLineNoise))
-					data = nptRemoveLineNoise(data,Args.RemoveLineNoise,sRate);
-				end
-				datam = mean(data);
-				[s,w,t,ps] = spectrogram(data-datam,Args.TFfftWindow, ...
-					Args.TFfftOverlap,Args.TFfftPoints,sRate);
-				% add to overall mean
-				% get columns of ps
-				psIdx = 1:size(ps,2);
-				ops(:,psIdx) = ops(:,psIdx) + ps;
-				opsCount(:,psIdx) = opsCount(:,psIdx) + 1;
-			end
-			imagesc(0:(Args.TFfftWindow-Args.TFfftOverlap):mIdx,0:(sRate/Args.TFfftPoints):(sRate/2),ops./opsCount)
-            set(gca,'Ydir','normal')
-		else
-			data = obj.data.analogData(idx);
-			if(~isempty(Args.RemoveLineNoise))
-				data = nptRemoveLineNoise(data,Args.RemoveLineNoise,sRate);
-			end
-			datam = mean(data);
-			spectrogram(data-datam,Args.TFfftWindow,Args.TFfftOverlap,Args.TFfftPoints, ...
-				sRate,'yaxis')
-		end
-	elseif(Args.TFWavelets)
-		% not fully completed yet
-    	cdata = nptRemoveLineNoise(obj.data.analogData',50,1000);
-    	cdata1 = cdata(idx);
-    	cdata1m = mean(cdata1);
-    	PlotFFT(cdata1-cdata1m,1000);
+%     % find indices for n-th trial
+% 	tIdx = obj.data.trialIndices(n,:);
+%     sRate = obj.data.analogInfo.SampleRate;
+%     swrInfo = obj.data.analogInfo.Swr;
+%     
+% 	if(length(tIdx)==2)
+% 		OldMarkerFormat = 1;
+% 	else
+% 		OldMarkerFormat = 0;
+%     end
+%     
+%     if(OldMarkerFormat)
+% 		idx = (tIdx(1)-(Args.PreTrial/1000*sRate)):tIdx(2);
+% 	else
+% 		idx = (tIdx(1)-(Args.PreTrial/1000*sRate)):tIdx(3);
+%     end
+%     
+%     if(Args.Rms)
+%         if (Args.PlotAllData)
+%             data = obj.data.analogRmsData;
+%         else
+%             data = obj.data.analogRmsData(idx,:);
+%         end
+%         sc = 200; %scaling for graphs
+%         numPlots = floor(size(data,2)/sc);
+%     else
+%         if (Args.PlotAllData)
+%             data = obj.data.analogData;
+%         else
+%             data = obj.data.analogData(idx,:);
+%         end
+%         sc = 1000;
+%         numPlots = floor(size(data,2)/sc);
+%     end
+%     if(~isempty(Args.RemoveLineNoise))
+% 			data = nptRemoveLineNoise(data,Args.RemoveLineNoise,sRate);
+%     end
+%     
+%     j = 1;
+%     while j < numPlots+1
+%         
+%         xi = (j-1)*200+1;
```

Comment:
Please avoid using constants in your code. Set them up as optional input arguments instead.